python/src/tests/test_response.py

```diff
@@ -36,222 +41,84 @@ def test_upload_returns_key_with_uuid_as_folder_when_uploading(
-        spy = mocker.spy(resp, "_upload")
-
-        resp.publish()
-        assert spy.call_count == 0
-
-    @mock.patch("boto3.client")
-    def test_publishing_multiple_short_files_results_in_no_s3_uploads(
-        self, mocked_client, mocker
-    ):
-        stubbed_client = botocore.session.get_session().create_client(
-            "s3", **config.BOTO_RESOURCE_KWARGS
-        )
-        stubber = Stubber(stubbed_client)
-        stubber.activate()
-
-        result = Result(
-                "a",
-                content_encoding="base64",
-                content_type="application/octet-stream",
-            )
-
-        resp = Response(self.request, result)
-        spy = mocker.spy(resp, "_upload")
-
-        resp.publish()
-        assert spy.call_count == 0
-
-    @mock.patch("boto3.client")
-    def test_publishing_a_short_file_results_in_inlined_response(
-        self, mocked_client, mocker
-    ):
-        stubbed_client = botocore.session.get_session().create_client(
-            "s3", **config.BOTO_RESOURCE_KWARGS
-        )
-        stubber = Stubber(stubbed_client)
-        stubber.activate()
-
-        result = Result(
-                "a",
-                content_encoding="base64",
-                content_type="application/octet-stream",
-            ),
-
-        resp = Response(self.request, result)
-        result = resp.publish()
-
-        assert "inline" in result
-        assert "s3-path" not in result
-
-    @mock.patch("boto3.client")
-    def test_publishing_a_long_file_reuslts_in_s3_path_response(
-        self, mocked_client, mocker
-    ):
-        stubbed_client = botocore.session.get_session().create_client(
-            "s3", **config.BOTO_RESOURCE_KWARGS
-        )
-        stubber = Stubber(stubbed_client)
-        stubber.activate()
-
-        result = Result(
-                "a" * 512 * 1024,
-                content_encoding="base64",
-                content_type="application/octet-stream",
-            )
-
-        resp = Response(self.request, result)
-        result = resp.publish()
-
-        assert "inline" not in result
-        assert "s3-path" in result
-
-    @mock.patch("boto3.client")
-    def test_old_requests_do_get_sent(self, mocked_client, mocker):
-        stubbed_client = botocore.session.get_session().create_client(
-            "s3", **config.BOTO_RESOURCE_KWARGS
-        )
-        stubber = Stubber(stubbed_client)
-        stubber.activate()
-
-        request = self.request
-        request["timeout"] = "2000-01-01 00:00:00"
-
-        result = Result(
-                "a",
-                content_encoding="base64",
-                content_type="application/octet-stream",
-            )
-
-        resp = Response(self.request, result)
-        result = resp.publish()
-
-        assert result
+        response_msg = resp._construct_response_msg()
+
+        assert response_msg["request"] == self.request
+        assert response_msg["response"]["cacheable"] == True
+
```

Comment:
Add assertion for response error (@cosa65 suggestion)

```suggestion
        assert response_msg["request"] == self.request
        assert response_msg["response"]["cacheable"] == True
        assert response_msg["response"]["error"] == False
```